### PR TITLE
Story Mode Roadmap for v0.7 and Future

### PR DESCRIPTION
**Pull request by Rusty-Box**
A clear roadmap for all things related to story mode.

### CODE REVIEW

File: Story-Mode-Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+This page aims to lay down the goals for story mode to the upcoming v0.7 release and future SuperTux versions.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:1]: story mode ***for***

File: Story-Mode-Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+5. [Story Summary (v0.7)](#story-summary-for-v07)
+6. [Future Story Draft (post-v0.7)](#story-summary-for-v07)
+
+---
+
+## Cutscenes
+
+**Part I: Improve current cutscene quality**
+
+- Make sure all cutscenes script are stored in their own separate .nut file.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:21]: ***scripts***

File: Story-Mode-Roadmap.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+- Add comments for things to be changed or subject to being updated once certain contents are added or are missing for the being.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:22]: ***time being*** (that sentence is overly complicated)

File: Story-Mode-Roadmap.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+- A cutscene after the Ghost Tree fight leading to Rooted Forest's new outro level
+
+---
+
+## Level Difficulty
+
+- Drastically improve the overall difficulty curve to prevent drastic changes in difficulty across levels. The first levels shall start off easy,
+  increasing slowly in difficulty with the last few levels and castle level being one of the hardest of a world.
+- When arriving at a subsequent world, the difficulty scales back down again, roughly around the mid-way point of the previous world's difficulty.
+- Insure levels are kept at a consistent length based on their position in the worldmap progression and story significance
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:37]: I think you mean ***Ensure***

File: Story-Mode-Roadmap.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  (see [Level-Design](https://github.com/SuperTux/supertux/wiki/Level-Design#size) for more details).
+
+## Level Progression
+
+- Teach mechanics in bite-sized chunks
+  - First level introduces base mechanics only, nothing too complex
+  - Second level is a first test of said mechanics
+  - Third level solely introduces swimming alone since it will be a more common and important mechanic further on in story mode
+  - Fourth  level introduces more "advanced" mechanics (i.e. switches, ladders, sliding)
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:46]: double space

File: Story-Mode-Roadmap.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+- Introduce powerups one at a time, spread out through story mode - in levels that require their abilities to be used so players get a basic grasp over what they offer
+  - Introduce Fire Flower in the first level along the other basic elements
+  - Introduce Air Flower in the first airborne level of the 1st world
+  - Introduce Earth Flower early in world 2 followed by the Ice Flower a couple level in (mid-way point at the latest)
+
+---
+
+## Art Direction
+
+It is very important to distinguish each world of Story Mode to ensure they don't feel the same and offer a fresh visual expiereince!
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:58]: ***experience***

File: Story-Mode-Roadmap.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+### Icy Island
+
+**Enviornmental Design**
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:62]: ***Environmental***

File: Story-Mode-Roadmap.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+![](images/concept_art_rustybox/world1_design_concept.png)
+
+Icy Island shall evouce a friendly and familiar presents in its environment and character design. Like a Winter Wonderland if you will.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:66]: ***evoke**
***presence***

File: Story-Mode-Roadmap.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+Tux knows this place best and mostly nothing shall be of big shock or surprise. The Crystal Mine is the only place that has slight feel
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:67]: has ***a*** slight feel

File: Story-Mode-Roadmap.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+**Creature Design**
+
+![](images/concept_art_rustybox/world1_creature_design.png)
+
+It is Tux's home island and thus all creatures he encounters here should not seem all too threatening. Simple shapes, friendly faces, an
+overall cutesy look. An exception to this is the Yeti with his more more animalistic look, being a brutish beast on the hunt for Tux.
+
+### Rooted Forest + Corrupted Forest
+
+**Enviornmental Design**
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:79]: ***Environmental***

File: Story-Mode-Roadmap.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+![](images/concept_art_rustybox/world2_design_concept.png)
+
+Rooted Forest, compared to the previous world, is an unfamiliar, strange, but also mysterious and magical place. It is much more vibrant
+and varied in color. Tux is now in a brand new world far from his home that has its own rules and eco-system.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:84]: ***ecosystem***

File: Story-Mode-Roadmap.md [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+Its creatures are designed to lean more into a more vicious and complex look. Some even resemble real animals, like Igel, Owl and Mole.
+The idea is to show how outlandish this whole new world feels to Tux. The design of Granito goes back to the design approach of Icy Island
+to immediately invoke a feeling of familiarity, indicating their friendly presence.
+
+Their corrupted counterparts have their designs twisted into lifeless husks, lacking all personality, having become uniform through their
+shared misery. Even the friendly looking Granito, Tux encountered before have now been turned into the mindless creatures.
+
+### Tropical Paradise
+
+**Enviornmental Design**
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:102]: see above

File: Story-Mode-Roadmap.md [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+
+As a direct opposite to Icy Island's calm and cold feel, Tropical Paradise will be a hot and aggressive place. The further Tux progresses
+the more dangerous the environment becomes, starting with a nice open beach setting, to a more covered jungle-beach hybrid into the deep
+dark thicket that is the main jungle area in the paradises center, littered in temple structures and totems.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:106]: ***littered with***

File: Story-Mode-Roadmap.md [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+
+**Creature Design**
+
+This place will have the most bizarre creature design out of all the worlds. Much more detailed, animalistic and violent. Toucan will be
+ironically be the most "friendly" and "harmless" looking creature of this world further showcasing his status of being a push-over, more
+show than an actual threat to Tux and his quest, unlike the Master Totem.
+
+### Nolok's Mountain Peak
+
+**Enviornmental Design**
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:116]: s. a.

File: Story-Mode-Roadmap.md [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+All creatures encountered here will be mechanical with barely any facial expression. Many of them are actually a re-implementation of older
+designs, like the original black Mr. Bomb and Jumpy design during the Milestone 1 development era (represented through Rusty Bomb + Hellbent
+and Metaljumpy) as well as the scrapped Mr. Rocket design during the early Milestone 2 development era.
+
+---
+
+## Story Summary (for v0.7)
+
+**Icy Island**
+  - Nolok, captures Penny during a picnic, using purple roots to knockout Tux instantly
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:133]: ***knock Tux out*** (I think)

File: Story-Mode-Roadmap.md [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+  - Nolok leaves behind a note leading him to a castle across the island
+  - Tux spends an entire day, starting mid-day, making his way to said castle and reaches the icy cave around night time
+  - By the time he reached the exit of the cave the sun begins rising
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:136]: to rise (I think)

File: Story-Mode-Roadmap.md [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+  - At this point the Yeti makes his first appearance keeping a close eye on Tux (Tux is unaware of the Yetis presence)
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:137]: At this point, <-- comma
Yeti's <--- apostrophe

File: Story-Mode-Roadmap.md [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+  - After reaching and traversing the castle, the yeti reveals himself to Tux, initiating the fight after Tux flees from him to an outside area
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:138]: capitalize "Yeti" (Check the rest of those instances yourself)

File: Story-Mode-Roadmap.md [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+---
+
+## Future Story Draft (post-v0.7)
+
+This section is an early and still unfinished draft of the continuation of story mode past Rooted Forest.
+Note that all you see here is but an idea of what is to come story-wise and may differ from whatever the final execution may be!
+
+**Tropical Paradise**
+  - Tux arrives at night
+  - Tux must swim across a tiny isle to the main island
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:165]: swimming *across* an isle seems weird b/c it's solid matter

**Review by tobbi — CHANGES_REQUESTED** [review]

**Review by Vankata453 — CHANGES_REQUESTED** [review]

**Comment by MatusGuy** [comment]
Gatdamn

**Comment by tobbi** [comment]
> Gatdamn

Gato damn?

File: Story-Mode-Roadmap.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobbi) [reply Story-Mode-Roadmap.md]: I think "subject to" without "a"

**Comment by MatusGuy** [comment]
> Gato damn?

goddamn. ☹☹☹

File: Story-Mode-Roadmap.md [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+**Nolok's Mountain Peak**
+  - Tux finds himself on top of a huge mountain
+  - There is a massive fortress in the center of the mountain's peak
+  - The entrance door is locked by five colored keys
+
+> If Tux has not all five NOLOK-keys collected:
+
+  - Nolok tells Tux to bring all the five hidden keys spread through out Icy Island, Rooted Forest and Tropical Paradise
+
+> If Tux has all five NOLOK-keys collected:
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:184]: NOLOK keys

File: Story-Mode-Roadmap.md [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+**Creature Design**
+
+![](images/concept_art_rustybox/world2_creature_design.png)
+
+Its creatures are designed to lean more into a more vicious and complex look. Some even resemble real animals, like Igel, Owl and Mole.
+The idea is to show how outlandish this whole new world feels to Tux. The design of Granito goes back to the design approach of Icy Island
+to immediately invoke a feeling of familiarity, indicating their friendly presence.
+
+Their corrupted counterparts have their designs twisted into lifeless husks, lacking all personality, having become uniform through their
+shared misery. Even the friendly looking Granito, Tux encountered before have now been turned into the mindless creatures.
```
[the remark this answers is not in the document]
Comment (tobbi) [reply Story-Mode-Roadmap.md:98]: friendly-looking

File: Story-Mode-Roadmap.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vankata453) [reply Story-Mode-Roadmap.md]: Right.

File: Story-Mode-Roadmap.md [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+
+**Environmental Design**
+
+![](images/concept_art_rustybox/world2_design_concept.png)
+
+Rooted Forest, compared to the previous world, is an unfamiliar, strange, but also mysterious and magical place. It is much more vibrant
+and varied in color. Tux is now in a brand new world far from his home that has its own rules and ecosystem.
+
+The Corrupted Forest segment shall invoke a feeling of uneasy-ness, confusion and even loneliness. The forest has lost its vibrant colors.
+Everything is gritty, desolate, cold, covered in this net of purple, almost organic looking roots everywhere you look.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:87]: covered in **a** net

File: Story-Mode-Roadmap.md [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+Their corrupted counterparts have their designs twisted into lifeless husks, lacking all personality, having become uniform through their
+shared misery. Even the friendly looking Granito, Tux encountered before have now been turned into the mindless creatures.
+
+### Tropical Paradise
+
+**Environmental Design**
+
+As a direct opposite to Icy Island's calm and cold feel, Tropical Paradise will be a hot and aggressive place. The further Tux progresses
+the more dangerous the environment becomes, starting with a nice open beach setting, to a more covered jungle-beach hybrid into the deep
+dark thicket that is the main jungle area in the paradises center, littered with temple structures and totems.
```
Comment (tobbi) [inline Story-Mode-Roadmap.md:106]: paradise's

File: Story-Mode-Roadmap.md (hunk 20 above)
Comment (Rusty-Box) [reply Story-Mode-Roadmap.md:184]: I think it does not matter in this case, does it not?

File: Story-Mode-Roadmap.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Rusty-Box) [reply Story-Mode-Roadmap.md]: "Thicket" is an actual word. "Ticket" makes no sense here.

File: Story-Mode-Roadmap.md [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+  - First level introduces base mechanics only, nothing too complex
+  - Second level is a first test of said mechanics
+  - Third level solely introduces swimming alone since it will be a more common and important mechanic further on in story mode
+  - Fourth  level introduces more "advanced" mechanics (i.e. switches, ladders, sliding)
+  - and so on...
+
+- Introduce powerups one at a time, spread out through story mode - in levels that require their abilities to be used so players get a basic grasp over what they offer
+  - Introduce Fire Flower in the first level along the other basic elements
+  - Introduce Air Flower in the first airborne level of the 1st world
+  - Introduce Earth Flower early in world 2 followed by the Ice Flower a couple level in (mid-way point at the latest)
```
[the remark this answers is not in the document]
Comment (Rusty-Box) [reply Story-Mode-Roadmap.md]: Didn't I do that though?

File: Story-Mode-Roadmap.md [hunk 25]
```diff
@@ -0,0 +1,188 @@
...
+
+**Icy Island:**
+  - Nolok captures Penny during a picnic, using purple roots to knock out Tux instantly
+  - Nolok leaves behind a note leading him to a castle across the island
+  - Tux spends an entire day, starting mid-day, making his way to said castle and reaches the icy cave around night time
+  - By the time he reached the exit of the cave the sun begins to rise
+  - At this point, the Yeti makes his first appearance, keeping a close eye on Tux (Tux is unaware of the Yeti's presence)
+  - After reaching and traversing the castle, the Yeti reveals himself to Tux, initiating the fight after Tux flees from him to an outside area
+  - When beat, the Yeti accidentally informs Tux on the crystal mines and a boat to use to sail up north
+  - The sun sets when Tux arrives at the crystal mines He starts sailing up north during the night
```
Comment (Vankata453) [inline Story-Mode-Roadmap.md]: Missing fulls top after "crystal mines".

File: Story-Mode-Roadmap.md [hunk 26]
```diff
@@ -0,0 +1,188 @@
...
+  - Tux arrives at night at a small beach isle
+  - Tux must swim from there to the main island
+  - More and more Totems appear on the island, the closer and closer Tux gets to the island's temple
+  - Toucan is found inside their throne room, admiring himself on a painting
+  - Toucan is infuriated by Tux's sudden appearance, leading him outside, initiating the fight
+  - When beat, Toucan flees the scene with Tux following suit, in hopes to finally find out about Nolok and Penny's whereabouts
+  - Tux follows Toucan to a hidden fort with a teleporter
+  - Nolok is on his way to activate the teleporter
+  - Upon noticing Tux, Nolok pushes Penny into the teleporter as he summons purple roots to block Tux's path, using the teleporter himself soon after
+  - Tux fights himself through the fort until he manages to reach and use the teleporter to follow suit
```
Comment (Vankata453) [inline Story-Mode-Roadmap.md]: Comma before "until".

**Comment by NickerAban** [comment]
0.7.0 already arrived? or 70

**Review by Vankata453 — APPROVED** [review]

**Review by mrkubax10 — APPROVED** [review]

**Pull request merged**